### PR TITLE
Stop libxml2 2.12 migration as it's unnecessary

### PR DESCRIPTION
**Pull request by isuruf**
See https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/634

### CODE REVIEW

**Comment by isuruf** [comment]
Let's wait for a day or two to see if there's fallout from the repodata patch. If not, let's merge.

**Comment by isuruf** [comment]
cc @hmaarrfk @conda-forge/libxml2

**Comment by hmaarrfk** [comment]
I think we need to merge otherwise feedstocks that haven't migrated, like https://github.com/conda-forge/libhwloc-feedstock/pull/72

will continue to build with the pin.


but the maintainers will have to know to actually update to 2.12 with something like:
https://github.com/conda-forge/libhwloc-feedstock/pull/73

File: recipe/migrations/libxml2212.yaml [hunk 1]
```diff
@@ -3,5 +3,5 @@ __migrator:
   kind: version
   migration_number: 1
 libxml2:
-- '2.12'
+- 2
```
Comment (isuruf) [inline recipe/migrations/libxml2212.yaml:6]: Let's keep this file for a couple of days and delete later.

Comment (hmaarrfk) [reply recipe/migrations/libxml2212.yaml:6]: ah ok. fine by me.

**Comment by hmaarrfk** [comment]
do we need to wait for the repodata patches to go through and stuff?

**Comment by isuruf** [comment]
Yes. Forgot about that.

**Comment by hmaarrfk** [comment]
lets just merge when the top messages says "1 hour ago" on github.

**Comment by hmaarrfk** [comment]
image

**Comment by h-vetinari** [comment]
Link to relevant [analysis](https://github.com/conda-forge/libxml2-feedstock/pull/111#issuecomment-1892217450)

**Pull request merged**